File: guides/micronaut-aws-parameter-store/src/test/java/example/micronaut/SsmAsyncClientFactory.java
/*
 * Copyright 2017-2024 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package example.micronaut;

import io.micronaut.context.annotation.BootstrapContextCompatible;
import io.micronaut.context.annotation.Factory;
import io.micronaut.context.annotation.Primary;
import jakarta.inject.Singleton;
import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.ssm.SsmAsyncClient;
import software.amazon.awssdk.services.ssm.SsmAsyncClientBuilder;
import java.net.URI;
import java.net.URISyntaxException;

@Factory
@BootstrapContextCompatible
class SsmAsyncClientFactory {
    private final SsmConfig config;

    SsmAsyncClientFactory(SsmConfig config) {
        this.config = config;
    }

    @Primary
    @Singleton
    SsmAsyncClient createSecretsManagerClient(SsmAsyncClientBuilder builder) {
        try {
            return builder
                    .endpointOverride(new URI(config.getSsm().getEndpointOverride()))
                    .credentialsProvider(
                            StaticCredentialsProvider.create(
                                    AwsBasicCredentials.create(config.getAccessKeyId(), config.getSecretKey())
                            )
                    )
                    .region(Region.of(config.getRegion()))
                    .build();
        } catch (URISyntaxException e) {
            throw new RuntimeException(e);
        }
    }
}

File: guides/micronaut-aws-parameter-store/kotlin/src/test/kotlin/example/micronaut/SsmAsyncClientFactory.kt
/*
 * Copyright 2017-2026 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package example.micronaut

import io.micronaut.context.annotation.BootstrapContextCompatible
import io.micronaut.context.annotation.Factory
import io.micronaut.context.annotation.Primary
import jakarta.inject.Singleton
import software.amazon.awssdk.auth.credentials.AwsBasicCredentials
import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider
import software.amazon.awssdk.regions.Region
import software.amazon.awssdk.services.ssm.SsmAsyncClient
import software.amazon.awssdk.services.ssm.SsmAsyncClientBuilder
import java.net.URI
import java.net.URISyntaxException

@Factory
@BootstrapContextCompatible
class SsmAsyncClientFactory(private val config: SsmConfig) {

    @Primary
    @Singleton
    fun createSecretsManagerClient(builder: SsmAsyncClientBuilder): SsmAsyncClient {
        try {
            return builder
                .endpointOverride(URI(config.ssm.endpointOverride!!))
                .credentialsProvider(
                    StaticCredentialsProvider.create(
                        AwsBasicCredentials.create(config.accessKeyId!!, config.secretKey!!)
                    )
                )
                .region(Region.of(config.region!!))
                .build()
        } catch (e: URISyntaxException) {
            throw RuntimeException(e)
        }
    }
}
